lib/cache.ml

```diff
@@ -0,0 +1,68 @@
+open Bos
+open Rresult
+open Astring
+
+let pp_header = Fmt.(styled `Blue (const string "==> "))
+
+let valid_remote_name = String.map (function '/' | '@' | ':' -> '-' | x -> x)
+
+let git_remote_exists_or_create ~remotes ~remote =
+  let remote_name = valid_remote_name remote in
+  match String.Set.mem remote_name remotes with
+  | true -> Ok ()
+  | false -> Exec.git_remote_add ~remote_url:remote ~remote_name
+
+let git_branch_exists_or_create ~output_dir ~remotes ~branch ~remote ~tag cache_dir =
+  let remote_name = valid_remote_name remote in
+  OS.Dir.with_current cache_dir
+    (fun () ->
+      OS.Path.exists Fpath.(v ".git" / "refs" / "heads" / branch) >>= function
+      | true ->
+          Logs.app (fun l ->
+              l "%a%aPulling sources for %a." pp_header ()
+                Fmt.(styled `Green string)
+                "[CACHED] "
+                Fmt.(styled `Cyan Fpath.pp)
+                output_dir );
+          Ok ()
+      | false -> (
+          Logs.app (fun l ->
+              l "%aPulling sources for %a." pp_header () Fmt.(styled `Cyan Fpath.pp) output_dir );
+          git_remote_exists_or_create ~remotes ~remote >>= function
+          | () -> Exec.git_fetch_to ~remote_name ~tag ~branch ) )
+    ()
+  |> R.join
+
+let cache_dir =
+  let ( / ) = Filename.concat in
+  let default_cache_dir () =
+    let os_cache_dir = if Sys.win32 then "Local Settings" / "Cache" else ".cache" in
```

Comment:
You are handling the windows case here but still using the `HOME` variable, which will not be set. I think you should use the variable `UserProfile` (to have the `Local Settings` directory) or `AppData`.

Reply:
To solve this I'm using the same strategy as dune:
https://github.com/ocaml/dune/blob/master/src/xdg/xdg.ml